### DISCUSSION TITLE
Jump to next file with `neo-tree` api

### DISCUSSION THREAD

**Discussion by hacker-DOM** [nvim-neo-tree/neo-tree.nvim#1114] [Q&A]
Hi there! I was wondering if it is possile to do the following in neo-tree

I use vim-unimpaired ]f and [f heavily, I even have them bound to J & K. It opens the next or previous file under alphabetic sorting.

The issue is, my `lsd` and `neo-tree` put directories first, while unimpaired does not, and there are a few other differences - I have some global ignores in `neo-tree` that get opened when using unimpaired (eg `.metadata.json`).

My question is whether it would be possible to create a Lua function that would use the `neotree` api to open the next file, as would be seen by neo-tree

**Comment by cseickel** [comment]
We do something similar with the prev/next git modified commands (mapped to `[g` / `]g`. There is a utility function that takes a list of file paths and returns them sorted in the order that they would be displayed in a tree structure:

https://github.com/nvim-neo-tree/neo-tree.nvim/blob/70ab62c8cc5d039766c039597c7c33a76d74fd11/lua/neo-tree/sources/filesystem/commands.lua#L141
